include/seqan3/search/algorithm/search.hpp

```diff
@@ -28,14 +28,42 @@ namespace seqan3
 
 /*!\brief Search a query or a range of queries in an index.
  * \tparam index_t    Must model seqan3::FmIndex.
- * \tparam queries_t  Must model std::ranges::RandomAccessRange over the index's alphabet.
- *                    a range of queries must additionally model std::ranges::ForwardRange.
+ * \tparam queries_t  Must model std::ranges::RandomAccessRange over the index's alphabet and std::ranges::SizedRange.
+ *                    A range of queries must additionally model std::ranges::ForwardRange and std::ranges::SizedRange.
  * \param[in] queries A single query or a range of queries.
  * \param[in] index   String index to be searched.
  * \param[in] cfg     A configuration object specifying the search parameters (e.g. number of errors, error types,
  *                    output format, etc.).
- * \returns An object modelling std::ranges::Range containing the hits (the type depends on the specification
-            in `cfg`), or `void` if an on_hit delegate has been specified.
+ * \returns
+ * One object of the following type if one query was passed and a `std::vector` containing objects of the following type
+ * for each query if a range of queries was passed:
+ * <ul>
+ *     <li>If seqan3::search_cfg::index_cursor was selected in the configuration:
+ *     <ul>
+ *         <li>A `std::vector<typename index_t::cursor_type>` containing index_cursors at the text positions were the
+ *             search was successful.</li>
+ *     </ul>
+ *     </li>
+ *     <li>If seqan3::search_cfg::text_position was selected in the configuration:
+ *     <ul>
+ *         <li>For **single** text indices: A `std::vector<size_t>` representing text positions where the search
+ *             was successful.</li>
+ *         <li>For indices over text **collections**: A `std::vector<std::pair<size_t, size_t>>` where the
+ *             first element of the `std::pair` specifies the text index in the collection and the second element
+ *             contains the position in that text of the collection.</li>
+ *     </ul>
+ *     </li>
+ *     <li>If an on_hit delegate has been configured:
```

Comment:
In general it does look nice, but I am wondering if it would not be more readable or easier to understand if you would provide a table with two columns: input and output and 4 rows for the 4 different cases. 
I also suggest to introduce a simple aggregate type to replace the pair. So we have a strong type with named accessors.
Something like: 
```cpp
struct text_position
{
    size_t collection_index;  //!< The index pointing to the respective text in the collection.
    size_t text_index //!< The index within the respective text.
};

But the naming can be adapted if their are better namings.

Of course I am still in favor to also return this type in case you submit only a single query, which is just a special case of the collection interface. So the user only needs to worry between index_curser and text_poisition. Or maybe call it text_cursor as well? 
```

Reply:
| text_layout | search_cfg::output | result of search()                                                                                                                                                              |
|-------------|--------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| single      | text_position      | A std::vector representing text positions where the search was successful.                                                                                                      |
| single      | index_cursor       | A std::vector containing index_cursors at the text positions where the search was successful.                                                                                   |
| collection  | text_position      | A std::vector> where the first element of the std::pair specifies the text index in the collection and the second element contains the position in that text of the collection. |
| collection  | index_cursor       | A std::vector containing index_cursors at the text positions where the search was successful.                                                                                   |

Something like this?

Reply:
Regarding replacing the `pair`: We want to do a search result struct anyway, so no need to do API  breaking changes in this PR?

Reply:
Yeah something like the table above.

> Regarding replacing the pair: We want to do a search result struct anyway, so no need to do API breaking changes in this PR?

No not in this PR. Sorry, I meant it for future consideration, but it was not so clear.

Reply:
Ok, I'll try the table....but I'd do it in html. Otherwise a line will be 500 long since we can't do linebreaks in markdown tables.

Reply:
Oh no really 😱

Reply:
https://stackoverflow.com/questions/11700487/newline-in-markdown-table suggests to just use `<br>` to force the line break. I think it is still better than the entire table in html?

Reply:
<img width="1034" alt="Screenshot 2019-07-18 at 14 15 06" src="https://user-images.githubusercontent.com/12967715/61456485-7b2fcf80-a966-11e9-8686-2ddfdc8ebe5f.png">

Reply:
> https://stackoverflow.com/questions/11700487/newline-in-markdown-table suggests to just use `<br>` to force the line break. I think it is still better than the entire table in html?

But I don't want to insert a line break in the cell content, I want to linebreak the row.
Markdown: 1 line =  1 row

Reply:
But wouldn’t the line break not automatically render the entire row larger?

Reply:
What I mean to say is that if you put a `<br>` into a markdown table, it will only enforce a line break in the row (if html is supported).

| header 1 | header 2 |
|-----------|-----------|
| content 1| very long content that is very long and would go on forever because there wont be any line break inserted unless I say so or the renderer thinks it's time for a line break|

vs

| header 1 | header 2 |
|-----------|-----------|
| content 1| very long content that is very long<br> and would go on forever because there<br> wont be any line break inserted<br> unless I say so or the renderer thinks<br> it's time for a line break|


But the row still has to be a single line, e.g. no

| header 1 | header 2 |
|-----------|-----------|
| content 1| very long content that is very long
and would go on forever because there
 wont be any line break inserted
unless I say so or the renderer thinks
 it's time for a line break|

In general. Github markdown is a bit special as that it still displays it as a table, but doxygen wouldn't.

Since I don't want lines with a length of 400 (markdown), the markdown tables are translated to html anyways and html tables currently look nicer (more padding than markdown tables and a link to something in the header doesn't have the same color as the header background), I'd like to leave it like this since we will change it once the result struct is implemented.

Reply:
ok, sorry, for bugging. Just so bad that easy things are never easy. But let's keep it then as html table. 👍